pac4j-core/src/main/java/org/pac4j/core/credentials/authenticator/LocalCachingAuthenticator.java

```diff
@@ -56,8 +59,30 @@ public void validate(final T credentials, final WebContext context) throws HttpA
             });
             credentials.setUserProfile(profile);
             logger.debug("Found cached credential. Using cached profile {}...", profile);
-        } catch (Exception e) {
-            throw new CredentialsException(e);
+        } catch (UncheckedExecutionException e) {
+            if (e.getCause() instanceof RuntimeException) {
+                throw (RuntimeException) e.getCause();
+            } else if (e.getCause() != null) {
```

Comment:
The cache implementation can throw one of these 3 exceptions to wrap the original ones (the cause) OR for one of its own error (no cause).
The point of this change is to extract the original cause (from delegate.validate) and throw it instead of the wrapper.
The only optimisation I can see is to extract the getCause null check and TechnicalException throwing into a method.
But I don't see what is redundant. If you have a better way of expressing that, please share :)